docs/core/extensions/best-practices-for-developing-world-ready-apps.md

```diff
@@ -89,6 +94,20 @@ This section describes the best practices to follow when developing world-ready
 
 1. Note that the uiCulture value can be set to the browser accept language.
 
+1. For applications that are distributed, allow zero-downtime updates (eg, Azure container applications), or similar you must plan for situations where there may be multiple instances of the application with different format rules or other culture data, most relevantly timezone rules.
+    - If your application deployment includes a database, remember that the database will have its own globalization rules.  In most cases you should avoid performing any globalization-related functions in the database.
+    - If your application deployment includes a client application or web frontend using client globalization resources, assume that the client resources differ from the resources available to your server.  Consider exclusively performing globalization functions on the client.
+
+## Recommendations for robust testing
+
+1. To make dependencies more explicit and testing potentially easier and parallelizable, you _SHOULD CONSIDER_ explicitly passing culture-relevant settings, such as `CultureInfo` parameters, to methods that perform formatting, and `TimeZoneInfo` to methods that work with dates and times.  You **MUST** use <xref:System.TimeProvider> or a similar type when retrieving the time.
+
+1. For most tests, you _SHOULD NOT_ explicitly validate the exact output of a given formatting operation or the exact offset of a timezone.  Formatting and timezone data may change at any time, and may differ between two otherwise identical instances of an operating system (and potentially different processes on the same machine).  Relying on an exact value makes tests brittle.
+    - Generally, validating that some output was received will be sufficient (eg, non-empty strings when formatting).
+    - For some data elements and formats, validating that the data parses to the input value may be used instead (roundtripping).  Care needs to be taken for cases where fields are dropped (eg, year for some date-related fields) or the value truncated or rounded (such as for floating-point output).
+    - If you have explicit requirements to validate all localized format output, you _SHOULD CONSIDER_ creating and using a custom culture during test setup.  For most simple cases this can be done by instantiating a `CultureInfo` object via `new CultureInfo(..)` and setting the `DateTimeFormat` and `NumberFormat` properties, and can be entirely stable.  For more complicated cases, subclassing the type allows overriding additional properties.  There are potential additional benefits to this, such as enabling [pseudolocalization](https://learn.microsoft.com/en-us/globalization/methodology/pseudolocalization) via resource files.
```

Comment:
```suggestion
    - If you have explicit requirements to validate all localized format output, you _should consider_ creating and using a custom culture during test setup. For most simple cases this can be done by instantiating a `CultureInfo` object with its constructor `new CultureInfo(..)` and setting the `DateTimeFormat` and `NumberFormat` properties. For more complicated cases, subclassing the type allows overriding additional properties. There are potential additional benefits to this, such as enabling [pseudolocalization](/globalization/methodology/pseudolocalization) with resource files.
```